tools/doc/generate.js

```diff
@@ -18,15 +19,15 @@ args.forEach(function(arg) {
     format = arg.replace(/^\-\-format=/, '');
   } else if (arg.match(/^\-\-template=/)) {
     template = arg.replace(/^\-\-template=/, '');
+  } else if (arg.match(/^\-\-node\-version=/)) {
+    node_version = arg.replace(/^\-\-node\-version=/, '');
```

Comment:
This really is the version number. The doc build scripts are setup based on the premise that the version of Node running them is the same version the docs are being built for, by `tools/doc/html.js` using `process.version` to output the version to the docs. Since that no longer works with these changes I added this option to propagate the version the docs are being built for from Make through to `html.js`.